minichain/contract.py

```
@@ -1,13 +1,30 @@
+
+class GasMeter:
+    def __init__(self, limit):
+        self.gas = limit
+        self.initial_gas = limit
+
+    def trace_calls(self, frame, event, arg):
+        frame.f_trace_opcodes = True
+        if event == 'opcode':
+            self.gas -= 1
+            if self.gas <= 0:
+                raise OutOfGasException("Out of gas!")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)